react-native/react/util/hello.js

```diff
@@ -0,0 +1,30 @@
+/* @flow */
+// Send helloIAm message to service
+
+import engine from '../engine'
+import keybase from '../constants/types/keybase_v1'
+
+import type {ClientDetails} from '../constants/types/flow-types'
+
+export default function (pid: number, desc: string, argv: Array<string>): Promise<void> {
+  const details: ClientDetails = {
+    pid,
+    desc,
+    version: __VERSION__, // eslint-disable-line no-undef
+    argv: argv,
+    clientType: keybase.config.ClientType.gui
+  }
+
+  return new Promise((resolve, reject) => {
+    engine.listenOnConnect('hello', () => {
+      engine.rpc('config.helloIAm', {details}, {}, (err, resp) => {
```

Comment:
Feels weird we skip the whole redux flow. Fine for now but seems like we might want to dispatch an action?